staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go

```diff
@@ -58,9 +60,16 @@ func NameIsDNS1035Label(name string, prefix bool) []string {
 }
 
 // ValidateTenantName can be used to check whether the given tenant name is valid.
-// Prefix indicates this name will be used as part of generation, in which case
-// trailing dashes are allowed.
-var ValidateTenantName = NameIsDNSLabel
+func ValidateTenantName(name string, prefix bool) []string {
```

Comment:
where is this used? 

for a function named Validatexxx, it's a bit weird to see it returns a list of string instead of boolean.

Reply:
This func will be called in creating/updating an object. It will be called in ValidateObject() in k8s.io/apimachinery/pkg/api/validation/objectmeta.go, which is used to validate the object data .

As to the functio Validatexxx, the return value is an aggregation of errors. It should be better to return []error instead of []string. But as all the funcs about validation in this file and all the consumers of these funcs are working with the return value of []string, I made my code consistent with them.